lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -8794,11 +8797,29 @@ namespace Js
                 }
                 else
                 {
+                    // compact the types array by moving non-null types
+                    // at the beginning.
+                    this->types[nonNullIndex++] = type;
                     isAnyTypeLive = true;
+#if DBG
+                    isGuardValuePresent = this->guard->GetValue() == reinterpret_cast<intptr_t>(type) ? true : isGuardValuePresent;
+#endif
                 }
             }
         }
+        if (nonNullIndex > 0)
+        {
+            memset((void*)(this->types + nonNullIndex), 0, sizeof(Js::Type*) * (EQUIVALENT_TYPE_CACHE_SIZE - nonNullIndex));
+        }
+        else if(guard->IsInvalidatedWhileSweeping())
+        {
+            // just mark this as actual invalidated since there are no types
+            // present
+            guard->Invalidate();
```

Comment:
It was added more from perf perspective in `CheckIfTypeIsEquivalent` where first thing that is done is
```c++
if (guard->GetValue() == 0)
{
  return false;
}
```
So if the guard was invalidated while sweeping and none of equiv types are alive, we can return right away from `CheckIfTypeIsEquivalent`. Without this, we would go further and fetch `equivTypes[0]` and that would be `nullptr` and that is when we would have returned.